hogben/tests/test_optimise.py

```diff
@@ -135,3 +173,58 @@ def test_underlayers_func():
 
     expected_result = -1.500100628
     np.testing.assert_allclose(result, expected_result, rtol=1e-06)
+
+
+@patch('hogben.optimise.sig_fig_round')
+@patch('hogben.optimise.Optimiser.optimise_parameters')
+@patch('hogben.optimise.scan_parameters')
+@patch('hogben.optimise.Fisher.from_sample', new=MockFisher)
+def test_optimise_parameters(mock_scan_parameters, mock_optimise_parameters,
+                             mock_sig_fig_round):
+    """
+    Runs the general `optimise_parameters` workflow, and makes sure the output
+    is handled correctly, and data visualization is called.
+    """
+    mock_optimise_parameters.return_value = ([0.5, 0.5], 2.0)
+    mock_scan_parameters.return_value = None
+    mock_sig_fig_round.return_value = 0.5
+
+    sample = MockSample()
+    angle_times = [1, 2, 3]
+    result = optimise_parameters(sample, angle_times, visualise=True)
+
+    # Check if the sample parameters were updated correctly
+    assert result.params[0].value == 0.5
+    assert result.params[1].value == 0.5
+
+    # Check whether the visualisation functions are called
+    sample.sld_profile.assert_called_once()
+    sample.reflectivity_profile.assert_called_once()
+    sample.reflectivity_profile.scan_parameters()
```

Comment:
should this have an assert associated?

Reply:
Yes, it's a bit of a shallow test but it should just check whether scan_parameters is called. It's actually completely off, it should be `mock_scan_parameters.assert_called_once()`